test/puppetlabs/trapperkeeper/services/status/status_service_test.clj

```diff
@@ -406,17 +406,28 @@
 (deftest status-status-test
   (testing "trapperkeeper-status registers its own status callback"
     (with-status-service
-     app []
-     (let [resp (http-client/get "http://localhost:8180/status/v1/services")]
-       (is (= 200 (:status resp)))
-       (is (= #{:status-service} (ks/keyset (parse-response resp true)))))
-     (let [resp (http-client/get "http://localhost:8180/status/v1/services/status-service?level=debug")]
-       (is (= 200 (:status resp)))
-       (let [body (parse-response resp true)]
-         (is (= {:detail_level "debug"
-                 :service_name "status-service"
-                 :service_status_version 1
-                 :service_version status-core/status-service-version
-                 :state "running"}
-                (dissoc body :status)))
-         (is (map? (get-in body [:status :experimental :jvm-metrics]))))))))
+      app []
+      (let [resp (http-client/get "http://localhost:8180/status/v1/services")]
+        (is (= 200 (:status resp)))
+        (is (= #{:status-service} (ks/keyset (parse-response resp true)))))
+      (let [resp (http-client/get "http://localhost:8180/status/v1/services/status-service?level=debug")]
+        (is (= 200 (:status resp)))
+        (let [body (parse-response resp true)]
+          (is (= {:detail_level "debug"
+                  :service_name "status-service"
+                  :service_status_version 1
+                  :service_version status-core/status-service-version
+                  :state "running"}
+                 (dissoc body :status)))
+          (is (map? (get-in body [:status :experimental :jvm-metrics])))))
+      (testing "and does so idempotently"
+        (let [get-status-status #(-> (get-service app :StatusService)
+                                     service-context
+                                     :status-fns
+                                     deref
+                                     (get status-core/status-service-name))]
+          (is (not (nil? (get-status-status))))
+          (tka/stop app)
+          (is (nil? (get-status-status)))
+          (tka/start app)
```

Comment:
TK would never call start here w/o calling init first; probably best for this test to use `restart` instead.  But... I also think that doing that would make this test pass even w/o the new logic you've added to the `stop` fn.